.goreleaser-nightly.yml

```diff
@@ -5,7 +5,7 @@ builds:
     id: nats-server
     binary: nats-server
     ldflags:
-      - -w -X github.com/nats-io/nats-server/v2/server.gitCommit={{.ShortCommit}}
+      - -w -X github.com/nats-io/nats-server/v2/version.GitCommit={{.ShortCommit}} -X 'github.com/nats-io/nats-server/v2/version.Version={{.Summary}}'
```

Comment:
In this case `{{.Summary}}` behaves like `git describe` and uses the previous tag to generate a version

https://goreleaser.com/customization/templates/